File: apps/watchos/VaiMarciaWatch Watch App/Domain/AudioClip.swift
import Foundation

/// Espelha conceitualmente a entidade de domínio `AudioClip` do app mobile (veja
/// apps/mobile/lib/features/audio_playback/domain). Swift puro — sem imports de SDK de plataforma.
struct AudioClip: Identifiable, Equatable, Codable {
    let id: String
    let categoryId: String
    let title: String
    let localFileName: String?
    let durationMs: Int
    var isFavorite: Bool = false

    var isCachedLocally: Bool { localFileName != nil && !(localFileName?.isEmpty ?? true) }
}

File: apps/wearos/app/src/main/java/com/togplay/vaimarcia/wear/domain/AudioClip.kt
package com.togplay.vaimarcia.wear.domain

/**
 * Espelha conceitualmente a entidade de domínio `AudioClip` do app mobile (veja
 * apps/mobile/lib/features/audio_playback/domain). Mantida livre de dependências para que
 * possa ser testada unitariamente sem classes do framework Android.
 */
data class AudioClip(
    val id: String,
    val categoryId: String,
    val title: String,
    val localFilePath: String?,
    val durationMs: Int,
    val isFavorite: Boolean = false,
) {
    val isCachedLocally: Boolean get() = !localFilePath.isNullOrBlank()
}
